dfusion_rust_core/src/models/util.rs

```diff
@@ -22,6 +22,12 @@ impl PopFromLogData for u16 {
     }
 }
 
+impl PopFromLogData for u32 {
+    fn pop_from_log_data(bytes: &mut Vec<u8>) -> Self {
+        U256::pop_from_log_data(bytes).as_u32().try_into().unwrap()
```

Comment:
Doesn't `as_u32` return a u32 already? What die `try_into` and `unwrap` add here?

Reply:
Good question, I will take a look into this!